File: scala/src/main/scala/io/github/configur8/Misconfiguration.scala
package io.github.configur8

/**
 * Thrown when a problem occurs in creating a Configuration or retrieving a Property
 */
class Misconfiguration(message: String) extends RuntimeException(message)

object Misconfiguration {
  def throwUp(message: String, name: String) = () => throw new Misconfiguration(message + " '" + name + "'")
}

File: java/src/main/java/io/github/configur8/Misconfiguration.java
package io.github.configur8;

public class Misconfiguration extends RuntimeException {

    public static void throwUp(String message, String name) {
        throw new Misconfiguration(message + " '" + name + "'");
    }

    public Misconfiguration(String message) {
        super(message);
    }
}
